crypto/evp_extra/evp_asn1.c

```diff
@@ -70,6 +70,14 @@
 #include "internal.h"
 #include "../dilithium/internal.h"
 
+// The function parse_key_type takes the algorithm cbs sequence |cbs| and
+// extracts the OID. The OID is then searched against ASN.1 methods for a method
+// with that OID. As the |OID| is read from |cbs| the buffer is advanced.
+// For the case of |NID_rsa| the method |rsa_asn1_meth| is returned.
+// For the case of |EVP_PKEY_PQDSA| the method |pqdsa_asn1.meth| is returned, as
+// the OID is not returned (and the |cbs| buffer is advanced) we return the OID
+// as |cbs|. (This allows the specific OID, e.g. NID_MLDSA65 to be parsed to
+// type specific decoding functions within the algorithm parameter.
```

Comment:
// as |cbs|. (This allows the specific OID, e.g. NID_MLDSA65 to be parsed by
// the type-specific decoding functions within the algorithm parameter.)

Reply:
added in [e419b99](https://github.com/aws/aws-lc/pull/1963/commits/e419b9944904e8e30496d5a250b696997bc7e3f0)